MarkBind/markbind
# Handle generated plantuml files in functional testcases

**What is the purpose of this pull request?**




**Overview of changes:**
Fixes #1937
Fixes #1130 (arguably, this is an "expedient" solution such that if we include more/adjust plantuml files in our test sites in the future, we will need to update the handling code accordingly to ignore them. Let me know if I should change the keyword here to leave that issue open)

Also:
- update a warning in our DG -> workflows to mention the caveat, just in case
- add a note about how to update plantuml files that are ignored

**Anything you'd like to highlight/discuss:**
~~Besides deleting the plantuml images and git ignore them, I realized I will also need to delete them from the `expected` folder again, before running the functional test. This is because these files which will still be generated after an `npm run updatetest` will affect testing.~~
The plantuml images will still be generated in the `expected` folder, in case devs want to look at them.

Wonder if we should do the same thing for the font files, as they appear to be problematic as well.
- at the moment: no

**Testing instructions:**
For docs update:
https://deploy-preview-1993--markbind-master.netlify.app/devguide/workflow#updating-and-writing-tests
![image](https://user-images.githubusercontent.com/41845017/183283264-a2bf2d62-017d-42a7-be80-535904492057.png)
![image](https://user-images.githubusercontent.com/41845017/183699755-4a133cb1-3392-4d94-807a-def82c10f61e.png)

For testing code:
- `npm run test` should succeed
- Running `npm run updatetest` should no longer show plantuml files as uncommitted changes, but those files should actually be generated in the `expected` folder  (because they are added to .gitignore)
- Re running `npm run test` should pass again, and those generated plantuml files should still appear in the `expected` folder

**Proposed commit message: (wrap lines at 72 characters)**
Handle generated plantuml files in test sites

Unrelated changes to png files (generated by plantuml) appear
when updating functional test cases.

Let's mention the need to ignore .png files (and binary files in general)
in our developer guide. Let's also update functional tests such that
some files can be ignored, to reduce the need to deal with 
unrelated changes.

This helps newcomers to know how to update functional tests
properly and reduce unnecessary dev work to maintain the test cases.



---

**Checklist:** :ballot_box_with_check:
